### ISSUE TITLE
Pre-training fast feedback with configurable low number of epochs.

### ISSUE THREAD

**Issue by pkuppens** [MIC-DKFZ/nnUNet#865]
Requirement/Request:
----
1. When I do my first training, I want to know within a few hours (instead of days), if training will give the expected outcome.
2. Some trained models are already sufficiently accurate on 10's of epochs, instead of 1000.

Solution:
Make the number of epochs a configuration parameter with default 1000 (like it is now). E.g. additional option -m 10 should only train for a maximum of 10 epochs.
Just this extra parameter gives confidence that other parts of the model training will behave similarly on later epochs ('ceteris paribus').

Rationale/Justification:
----
Some errors in the setup of the training may only show up after the training. For a faster feedback loop, it is better to have feedback within minutes/hours instead of (multiple) days.

Examples (both occurred to me, on separate events):
1. input image was RGB (though gray) instead of intensity
2. label images looked like 0-1 masks, but actually were 0-255, with occasional values in between because of anti-aliasing(?!)

In my current configuration, an epoch takes ~200 seconds. For 1000 epochs this would be ~2.3 days per fold, so 2 weeks in total.

Both errors were obvious in hindsight but would have cost me weeks (each) if I would not have started pre-training with 10 epochs only, which cost me about 3 hours.

Comment:
----
The technical implementation does not appear too difficult, however, I'm not certain about implicit assumptions and their consequences, like learning rates that depend on the number of epochs.

**Comment by FabianIsensee** [comment]
I don't want to make this a configurable parameter because it interfers with nnU-Net's defaults and people might end up playing around with it even though they shouldn't. Also, nnU-Net saves results by class name, so it would no longerbe clear how many epochs were actually used for each training.

If you want to have a small number of epochs to test things, take a look at the `nnUNetTrainerV2_5epochs` trainer. It doesn't save checkpoints, it doesn't run validation. But it gives you a log file and a progress.png to look at.

You can also just write your own it you so choose. The easiest variant would be:
```python
class nnUNetTrainerV2_10epochs(nnUNetTrainerV2):
    def __init__(self, plans_file, fold, output_folder=None, dataset_directory=None, batch_dice=True, stage=None,
                 unpack_data=True, deterministic=True, fp16=False):
        super().__init__(plans_file, fold, output_folder, dataset_directory, batch_dice, stage, unpack_data,
                         deterministic, fp16)
        self.max_num_epochs = 10
```

It takes care of lr schedule etc internally.

Given a new dataset, I usually just run a regular training locally for a couple of epochs and cancel it manualy as soon as I see whether it's working or not. No need for a separate trainer imo

Best,
Fabian

**Comment by Karol-G** [comment]
Closing this issue as it seems solved. Feel free to open the issue again, if you have further questions.
